base/modulus_remainder.cc

```diff
@@ -0,0 +1,232 @@
+#include "base/modulus_remainder.h"
+
+// Based on https://github.com/halide/Halide/blob/main/src/ModulusRemainder.cpp.
+namespace slinky {
+
+namespace {
+
+int64_t div_imp(int64_t a, int64_t b) {
```

Comment:
We already have https://github.com/dsharlet/slinky/blob/5955e84b4faf77f206c400bd9818f5ff852be981/base/arithmetic.h#L16 and related functions in that file.